File: ManualProg.Api/Features/Users/Responses/UserResponse.cs
using ManualProg.Api.Data.Users;

namespace ManualProg.Api.Features.Users.Responses;

public record UserResponse
{
    public required Guid Id { get; set; }
    public required string Username { get; set; }
    public required UserRole Role { get; set; }
    public ProfileData? Profile { get; set; }
    public required DateTime Created { get; set; }

    public record ProfileData
    {
        public required Guid Id { get; set; }
        public required string Name { get; set; }
    }
}

File: ManualProg.App/src/app/users/data-access/responses/user-response.ts
import { UserRole } from "../../../shared/enums/user-role";

export interface IUserResponse {
  id: string;
  username: string;
  role: UserRole;
  profile: IUserResponseProfileData | null;
  created: Date;
}

export interface IUserResponseProfileData {
  id: string;
  name: string;
  hasImage: boolean;
}
